dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIPromptExecutionSettings.cs

```diff
@@ -156,6 +156,26 @@ public static OpenAIPromptExecutionSettings FromRequestSettings(PromptExecutionS
         throw new ArgumentException($"Invalid request settings, cannot convert to {nameof(OpenAIPromptExecutionSettings)}", nameof(executionSettings));
     }
 
+    /// <summary>
+    /// Create a new settings object with the values from another settings object.
+    /// </summary>
+    /// <param name="executionSettings">Template configuration</param>
+    /// <param name="defaultMaxTokens">Default max tokens</param>
+    /// <returns>An instance of OpenAIPromptExecutionSettings</returns>
+    public static OpenAIPromptExecutionSettings FromRequestSettingsWithData(PromptExecutionSettings? executionSettings, int? defaultMaxTokens = null)
+    {
+        var requestSettings = FromRequestSettings(executionSettings, defaultMaxTokens);
+
+        if (requestSettings.StopSequences?.Count == 0)
+        {
+            // Azure OpenAI WithData API does not allow to send empty array of stop sequences
+            // Gives back "Validation error at #/stop/str: Input should be a valid string\nValidation error at #/stop/list[str]: List should have at least 1 item after validation, not 0"
+            requestSettings.StopSequences = null;
```

Comment:
Should this be fixed in the OPenAI specific version of PromptExecutionSettings so that it applies to all calls to OpenAI?

Reply:
The normal APIs (Non WithData) validation don't complain about the empty array.

I'm applying this change to the WithData only to minimize any impact on the current code.

![image](https://github.com/microsoft/semantic-kernel/assets/19890735/c04c7134-2610-4564-b5e2-4f4e5bf2ed76)

Reply:
Analyzing further, in the ClientBase, using the Non WithData endpoints, Azure SDK model don't have nullable StopSequences for its `ChatCompletionsOptions` class.

![image](https://github.com/microsoft/semantic-kernel/assets/19890735/d836adae-d9bd-4209-9a4b-5f8965cc26b6)